packages/odyssey-react-mui/src/labs/SideNav.tsx

```diff
@@ -0,0 +1,601 @@
+        )),
+      })),
+    [sideNavItems],
+  );
+
+  const sideNavCollapseHandler = useCallback(() => {
+    setSideNavCollapsed(!isSideNavCollapsed);
+    onCollapse && onCollapse();
+  }, [isSideNavCollapsed, setSideNavCollapsed, onCollapse]);
+
+  return (
+    <Box
+      sx={{
+        display: "flex",
+        height: "100vh",
+      }}
+    >
+      <SideNavCollapsedContainer
+        tabIndex={0}
+        odysseyDesignTokens={odysseyDesignTokens}
+        isSideNavCollapsed={isSideNavCollapsed}
+        onClick={() => setSideNavCollapsed(!isSideNavCollapsed)}
+        onKeyDown={(event) => {
+          (event.key === "Enter" || event.code === "Space") &&
+            setSideNavCollapsed(!isSideNavCollapsed);
+        }}
+      >
+        <ExpandLeftIcon
+          sx={{
+            margin: `0 ${odysseyDesignTokens.Spacing1}`,
+            fontSize: "1em",
+          }}
+        />
+      </SideNavCollapsedContainer>
+      <SideNavContainer
+        odysseyDesignTokens={odysseyDesignTokens}
+        isSideNavCollapsed={isSideNavCollapsed}
+      >
+        <Box
+          sx={{
+            position: "sticky",
+            top: 0,
+          }}
+        >
+          <SideNavHeader
+            navHeaderText={navHeaderText}
+            isCollapsible={isCollapsible}
+            onCollapse={sideNavCollapseHandler}
+          />
+        </Box>
+        <Box
+          sx={{
+            flex: 1,
+            overflowY: "auto",
+          }}
+        >
+          <SideNavListContainer>
+            {processedSideNavItems?.map((item) => {
+              const {
+                id,
+                label,
+                isSectionHeader,
+                startIcon,
+                children,
+                isDefaultExpanded,
+                isDisabled,
+              } = item;
+
+              if (isSectionHeader) {
+                return (
+                  <SectionHeader
+                    id={id}
+                    key={id}
+                    odysseyDesignTokens={odysseyDesignTokens}
+                  >
+                    {label}
+                  </SectionHeader>
+                );
+              } else if (children) {
+                return (
+                  <SideNavListItemContainer
+                    id={id}
+                    key={id}
+                    odysseyDesignTokens={odysseyDesignTokens}
+                  >
+                    <NavAccordion
+                      label={label}
+                      isDefaultExpanded={isDefaultExpanded}
+                      startIcon={startIcon}
+                      isDisabled={isDisabled}
+                    >
+                      <SideNavListContainer id={id + "-list"}>
+                        {children}
+                      </SideNavListContainer>
+                    </NavAccordion>
+                  </SideNavListItemContainer>
+                );
+              } else {
+                return <SideNavItemContent key={item.id} {...item} />;
```

Comment:
What is the `else` case? When would end up here? Either we need to document that with a comment or better yet, add it as an `else if` and be explicit about what we're checking.

Reply:
else is essentially the default nav item case, i.e., item is not a section header or a parent (accordion) with children